### PR TITLE
Remove text mesh pro preprocessor directive

### PR DESCRIPTION
**Pull request by chrisfromwork**
Having different compiled binaries that support text mesh pro vs don't support text mesh pro seems like overkill. For now, we can remove the text mesh pro pre processor directive to ease our compiled unity binary setup process. This should make our automation/nuget setup simpler for binary distribution of spectator view logic.

### CODE REVIEW

File: samples/Build2019Demo.Unity/ProjectSettings/ProjectSettings.asset [hunk 1]
```diff
@@ -627,21 +627,24 @@ PlayerSettings:
   webGLLinkerTarget: 1
   webGLThreadsSupport: 0
   scriptingDefineSymbols:
-    1: SPATIALALIGNMENT_ASA;STATESYNC_TEXTMESHPRO
```
Comment (fibann) [inline samples/Build2019Demo.Unity/ProjectSettings/ProjectSettings.asset]: Are there any drawbacks with compiling the TextMeshPro bits unconditionally?

Comment (chrisfromwork) [reply samples/Build2019Demo.Unity/ProjectSettings/ProjectSettings.asset]: 1) It introduces a dependency on the text mesh pro unity package.
2) Folks consuming a different version of text mesh pro may have problems.

It's a bit unclear right now who uses Text Mesh Pro vs who doesn't. I think a more viable solution would be to move these classes into a separate dll with the text mesh pro dependency if folks voice that they really can't consume the Text Mesh Pro dependency.

I also think that the regressions caused by having these preprocessor definitions and the lack of discoverability around enabling/disabling these preprocessor definitions advocates for not keeping them.

Comment (chrisfromwork) [reply samples/Build2019Demo.Unity/ProjectSettings/ProjectSettings.asset]: technically right now the spectator view asmdef already has a text mesh pro dependency, so this preprocessor directive isn't helping avoid a text mesh pro dependency.

File: src/SpectatorView.Unity/Assets/SpectatorView.Tests/Microsoft.MixedReality.SpectatorView.Tests.msb4u.csproj.meta [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ScriptedImporter:
+  fileIDToRecycleName:
+    11400000: Microsoft.MixedReality.SpectatorView.Tests.msb4u
+  externalObjects: {}
+  userData: 
+  assetBundleName: 
+  assetBundleVariant: 
+  script: {fileID: 11500000, guid: 0e4dca1acba66fd478b3854f09a32ec7, type: 3}
+  buildEngine: 0
+  profiles: []
```
Comment (matthejo) [inline src/SpectatorView.Unity/Assets/SpectatorView.Tests/Microsoft.MixedReality.SpectatorView.Tests.msb4u.csproj.meta]: Seems like these might be the same .meta files that probably should be excluded by .gitignore?

Comment (chrisfromwork) [reply src/SpectatorView.Unity/Assets/SpectatorView.Tests/Microsoft.MixedReality.SpectatorView.Tests.msb4u.csproj.meta]: I'll ask andrei about this and will remove them as needed in a follow up review.

**Pull request merged**